cpp/src/centrality/betweenness_centrality.cu

```diff
@@ -26,10 +26,287 @@
 
 #include <gunrock/gunrock.h>
 
+#include "betweenness_centrality.cuh"
+
 namespace cugraph {
+namespace detail {
+
+template <typename VT, typename ET, typename WT, typename result_t>
+void BC<VT, ET, WT, result_t>::setup()
+{
+  // --- Set up parameters from graph adjList ---
+  number_of_vertices = graph.number_of_vertices;
+  number_of_edges    = graph.number_of_edges;
+  offsets_ptr        = graph.offsets;
+  indices_ptr        = graph.indices;
+}
+
+template <typename VT, typename ET, typename WT, typename result_t>
+void BC<VT, ET, WT, result_t>::configure(result_t *_betweenness,
+                                         bool _normalized,
+                                         bool _endpoints,
+                                         WT const *_weights,
+                                         VT const *_sources,
+                                         VT _number_of_sources)
+{
+  // --- Bind betweenness output vector to internal ---
+  betweenness       = _betweenness;
+  normalized        = _normalized;
+  endpoints         = _endpoints;
+  sources           = _sources;
+  number_of_sources = _number_of_sources;
+  edge_weights_ptr  = _weights;
+
+  // --- Working data allocation ---
+  ALLOC_TRY(&distances, number_of_vertices * sizeof(VT), nullptr);
+  ALLOC_TRY(&predecessors, number_of_vertices * sizeof(VT), nullptr);
+  ALLOC_TRY(&sp_counters, number_of_vertices * sizeof(double), nullptr);
+  ALLOC_TRY(&deltas, number_of_vertices * sizeof(result_t), nullptr);
+
+  // --- Get Device Information ---
+  CUDA_TRY(cudaGetDevice(&device_id));
+  CUDA_TRY(cudaDeviceGetAttribute(&max_grid_dim_1D, cudaDevAttrMaxGridDimX, device_id));
+  CUDA_TRY(cudaDeviceGetAttribute(&max_block_dim_1D, cudaDevAttrMaxBlockDimX, device_id));
+
+  // --- Confirm that configuration went through ---
+  configured = true;
+}
+template <typename VT, typename ET, typename WT, typename result_t>
+void BC<VT, ET, WT, result_t>::clean()
+{
+  ALLOC_FREE_TRY(distances, nullptr);
```

Comment:
Once you remove `ALLOC_TRY` and get these pointers from the associated vectors, these `ALLOC_FREE_TRY` calls can be removed.

Reply:
Updated in 4bdf822, removed the `clean` function from the BC class.